### PR TITLE
feat: adding llvm source based coverage

### PR DESCRIPTION
**Pull request by jrmncos**
#1174 
Adding coverage to all directories declared in members excepting /examples
The changes were tested locally and the file generated by llvm-cov (llvm-cov-output.lcov) can be parsed as html and it's displaying the coverage correctly. Also the step Generate code coverage is running successfully.

As a final step to fully test this change is to setup the variable ${{ secrets.CODECOV_TOKEN }} in the repo settings. More info here: https://docs.codecov.com/docs/quick-start#getting-started If someone can set that variable it would be great.
I did it in my forked repo, and I received an email with something like `Once you merge this PR into your default branch, you're all set! Codecov will compare coverage reports and display results in all future pull requests.` 

BTW, I have been dedicating most part of my time and effort to test the directory /examples with the tool wasmcov #1233 . I was able to generate the .profdata files for each directory, and maybe the next step it's to merge it and to integrate it with the changes of this PR. I'm working with @bbarwik was helping me to integrate the tool and I'm waiting for the next release that fix some bugs.

I would like to close this one and to continue working on #1233 with an accurate plan to ingrate /examples directory

### CODE REVIEW

**Comment by jrmncos** [comment]
@frol this is ready for review :)

File: .github/workflows/coverage.yml
Comment (frol) [inline .github/workflows/coverage.yml:24]: It will take too much time on every CI run. Can we use Linux as a host? It will be much faster to install it there.

**Review by frol — CHANGES_REQUESTED** [review]
@jrmncos Could you, please, also provide a link for a CI test run on your fork? It would be great to see that it fully works before we merge it into the master (I'd like to avoid us debugging CI in production)

File: .github/workflows/coverage.yml
Comment (jrmncos) [reply .github/workflows/coverage.yml:24]: in this repo: https://github.com/near/near-sdk-rs/actions/runs/10295407665/job/28495033969?pr=1236
in my fork: https://github.com/jrmncos/near-sdk-rs/actions/runs/10295407683/job/28495034208?pr=1

it's takes ~8 approximately. Let's try with Linux to reduce it

Comment (jrmncos) [reply .github/workflows/coverage.yml:24]: hey @frol I tried with ubuntu, and the longer step, that is 'Generate code coverage' it takes 7m:48seg https://github.com/near/near-sdk-rs/actions/runs/10360531522/job/28679132518
and with mac os 5m 36s
https://github.com/near/near-sdk-rs/actions/runs/10360643219/job/28679480453?pr=1236\

Even with the installation of Brew the total time of macos is still less. Also, the longer step in the GitHub actions pipelines it's Test core and it takes > 8m (this is the total time of coverage)

I tried using    - uses: Swatinem/rust-cache@v1
 as cache but has not improved the total time

**Review by frol — APPROVED** [review]
<10 minutes for code-coverage is not that bad. Let's see it in action!

@race-of-sloths invite

**Comment by race-of-sloths** [comment]
@frol Thank you for calling!
    
@jrmncos Thank you for the contribution! Join [Race of Sloths](https://race-of-sloths.com) by simply mentioning me in your comment/PRs description and start collecting Sloth Points through contributions to open source projects.

<details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/jrmncos)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by race-of-sloths** [comment]
🥁  __Score it!__

@frol, please score the PR with `@race-of-sloths score [1/2/3/5/8/13]`. The contributor deserves it.
If no scoring is provided within 24 hours, this PR will be scored as 2 🦥

**Pull request merged**
